From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Fri, 2 Oct 2026 20:42:23 +0200
Subject: [PATCH] Fix build with Rust 1.98.0+

error: invalid definition of the runtime `close` symbol used by the standard library
  --> src/vkeyboard.rs:49:9
   |
49 |         fn close(fd: u32);
   |         ^^^^^^^^^^^^^^^^^^
   |
   = note: expected `unsafe extern "C" fn(i32) -> i32` (for the current target)
           found    `unsafe extern "C" fn(u32)`
   = help: either fix the signature or remove any attributes like `#[unsafe(no_mangle)]`, `#[unsafe(export_name = "close")]`, or `#[link_name = "close"]`
   = note: `#[deny(invalid_runtime_symbol_definitions)]` on by default
---
 src/vkeyboard.rs | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/vkeyboard.rs b/src/vkeyboard.rs
index 886ce949628c..c208ad8a3883 100644
--- a/src/vkeyboard.rs
+++ b/src/vkeyboard.rs
@@ -39,14 +39,14 @@ pub mod c {
     impl Drop for KeyMap {
         fn drop(&mut self) {
             unsafe {
-                close(self.fd as u32);
+                close(self.fd as i32);
             }
         }
     }
 
     extern "C" {
         // From libc, to let KeyMap get deallocated.
-        fn close(fd: u32);
+        fn close(fd: i32) -> i32;
 
         pub fn eek_virtual_keyboard_v1_key(
             virtual_keyboard: ZwpVirtualKeyboardV1,
